core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for learner answer info.
+ */
+
+require(
+  'pages/exploration-player-page/services/answer-classification.service.ts');
+require('domain/statistics/LearnerAnswerDetailsBackendApiService.ts');
+
+angular.module('oppia').factory('LearnerAnswerInfoService', [
+  'AnswerClassificationService', 'LearnerAnswerDetailsBackendApiService',
+  'INTERACTION_IDS_WITHOUT_ANSWER_DETAILS',
+  function(
+      AnswerClassificationService, LearnerAnswerDetailsBackendApiService,
+      INTERACTION_IDS_WITHOUT_ANSWER_DETAILS) {
+    var submittedAnswerInfoCount = 0;
+    var currentEntityId = null;
+    var stateName = null;
+    var interactionId = null;
+    var currentAnswer = null;
+    var currentInteractionRulesService = null;
+    var canAskLearnerForAnswerInfo = false;
+    var visitedStates = [];
+    var probabilityIndexes = {
+      // The probability that a request for explanation of the answer that is
+      // submitted by the learner. There are three different probabilities
+      // based on the outcome of the answer.
+      // The probability index when the outcome is equal to the default outcome
+      // for an interaction.
+      type_a: 0.25,
```

Comment:
Why snake_case rather than the usual lowerCamelCase?

Reply:
Done